data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -368,15 +368,32 @@ fi
 
 echo "Starting cluster-bootstrap..."
 
-if [ ! -f cb-bootstrap.done ]
-then
-    bootkube_podman_run \
+run_cluster_bootstrap() {
+	bootkube_podman_run \
         --rm \
         --volume "$PWD:/assets:z" \
         --volume /etc/kubernetes:/etc/kubernetes:z \
         "${CLUSTER_BOOTSTRAP_IMAGE}" \
         start --tear-down-early=false --asset-dir=/assets --required-pods="${REQUIRED_PODS}"
+}
+
+gather_bootstrap_on_failure() {
+	/usr/local/bin/installer-gather.sh --id 'api-server-temp'
+	gunzip -k /tmp/artifacts-api-server-temp/resources/openapi.json.gz
+	if [ ! -s '/tmp/artifacts-api-server-temp/resources/openapi.json' ]
+	then
+		rm -rf /tmp/artifacts-api-server-temp
+		rm $HOME/log-bundle-api-server-temp.tar.gz
+	else
+		mv $HOME/log-bundle-api-server-temp.tar.gz $HOME/log-bundle-api-server.tar.gz
+	fi
+}
 
+
+if [ ! -f cb-bootstrap.done ]
+then
+	trap 'gather_bootstrap_on_failure' ERR
```

Comment:
Setting the trap means that gather_bootstrap_on_failure will run when there are errors in subsequent steps of the script. That is not terrible, but it is not quite what we want. Can we capture the result of the podman run instead of trapping?

Reply:
I did try podman logs/history (which I feel might be hard to parse) and tried to reset traps (didn't work).  Any other ideas?